File: leetcode/100-999/145. Binary Tree Postorder Traversal/solution_1.js

/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */

/**
 * @param {TreeNode} root
 * @param {number[]} tree
 * @return {void}
 */

function walkTree(tree, root)
{
   if (root)
   {
       walkTree(tree, root.left);
       walkTree(tree, root.right);
       tree.push(root.val);
   }
}

/**
* @param {TreeNode} root
* @return {number[]}
*/

var postorderTraversal = function(root) 
{
   const tree = []
   walkTree(tree, root)
   return tree
};

File: leetcode/100-999/145. Binary Tree Postorder Traversal/solution_1.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution 
{
    private List<int> treeVal = new List<int>();

    public IList<int> PostorderTraversal(TreeNode root) 
    {
        if (root != null)
        {
            PostorderTraversal(root.left);
            PostorderTraversal(root.right);
            treeVal.Add(root.val);
        }

        return treeVal;
    }
}